framework/src/problems/FEProblemBase.C

```diff
@@ -464,6 +464,7 @@ FEProblemBase::FEProblemBase(const InputParameters & parameters)
             ? getParam<bool>("error_on_jacobian_nonzero_reallocation")
             : _app.errorOnJacobianNonzeroReallocation()),
     _ignore_zeros_in_jacobian(getParam<bool>("ignore_zeros_in_jacobian")),
+    _preserve_matrix_sparsity_pattern(true),
```

Comment:
If ignore-zeros is a user-facing option, I would like this to be user-facing too. Also, I think for FV, depending on the coupling, many entries get condensed out. For example, if the material properties depend on cell temperature alone then all the coupling with the neighbor temperatures will be condensed out. Also, this could be really confusing, if someone notices that this is true and petsc still condenses the entries out. Plus, this is FV specific at the moment. I think maybe adding some documentation to this to a design page would make these clear.

Reply:
why? do you have a use for it? Having parameters we dont use does not help anyone.

Reply:
Especially since other kernels might not be implementing this correctly at this point. It's not enforced, we can have this set to true and kernels from 3 years ago don't preserve sparsity pattern

Reply:
Okay, I suppose this is temporary in a sense that when we migrate the turbulence and other routines to the linear solver, we can remove this duality in the nonlinear FV kernels. And then we might be able to remove this too.

Reply:
I'd be fine with removing it if the need goes away.